care/facility/api/viewsets/patient.py

```diff
@@ -134,3 +135,50 @@ def list(self, request, *args, **kwargs):
 
         """
         return super(FacilityPatientStatsHistoryViewSet, self).list(request, *args, **kwargs)
+
+
+class PatientSearchViewSet(viewsets.ModelViewSet):
+    http_method_names = ["get"]
+    queryset = PatientSearch.objects.all()
+    serializer_class = PatientSearchSerializer
+    permission_classes = (IsAuthenticated, DRYPermissions)
+
+    def get_queryset(self):
+        if self.action != "list":
+            return super(PatientSearchViewSet, self).get_queryset()
+        else:
+            serializer = PatientSearchSerializer(data=self.request.query_params, partial=True)
+            serializer.is_valid(raise_exception=True)
+
+            search_keys = ["date_of_birth", "year_of_birth", "phone_number"]
+            search_fields = {
+                key: serializer.validated_data[key] for key in search_keys if serializer.validated_data.get(key)
+            }
+            if not search_fields:
+                raise serializers.ValidationError(
+                    {"detail": [f"None of the search keys provided. Available: {', '.join(search_keys)}"]}
+                )
+
+            if not self.request.user.is_superuser:
+                search_fields["state_id"] = self.request.user.state_id
+            return self.queryset.filter(**search_fields)
+
+    def retrieve(self, request, *args, **kwargs):
+        raise NotImplementedError()
+
+    def list(self, request, *args, **kwargs):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Useless super delegation in method 'list'](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5332931)